src/components/ui/cart/cart.lineitem.jsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Link } from 'react-router-dom';
+import { login } from '../../../utils/AuthService.js';
+
+var Config = require('Config')
+
+class CartLineItem extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            quantity: this.props.item.quantity
+        }
+        this.handleQuantityChange = this.handleQuantityChange.bind(this);
+        this.handleRemoveBtnClicked = this.handleRemoveBtnClicked.bind(this);
+    }
+    handleQuantityChange(event) {
+        event.preventDefault();
+        let newQuantity = event.target.value;
+
+        login().then(() => {
+            this.setState({ quantity: newQuantity }, () => {
+                fetch(this.props.item.self.href,
+                    {
+                        method: 'put',
+                        headers: {
+                            'Content-Type': 'application/json',
+                            'Authorization': localStorage.getItem(Config.cortexApi.scope + '_oAuthToken')
+                        },
+                        body: JSON.stringify({
+                            quantity: this.state.quantity
+                        })
+                    })
+                    .then(() => {
+                        this.props.handleQuantityChange()
```

Comment:
this is going to re-call the same function. What's the reason for this?

Reply:
no, this calls back to the parent component handler. this is so that upon changing the quantity, and issuing the put request to modify the contents of the cart, there needs to be a callback to the page to re-load the data (since the cartdata is set at the page component and passed along)